src/components/ComponentsListView/ComponentFilter.tsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import {
+  Select,
+  SelectVariant,
+  SelectOption,
+  SelectGroup,
+  ToolbarChip,
+  ToolbarGroup,
+  ToolbarFilter,
+} from '@patternfly/react-core';
+import { FilterIcon } from '@patternfly/react-icons/dist/esm/icons/filter-icon';
+
+export const NEEDS_MERGE_FILTER_ID = 'needs-merge';
+export const FAILED_STATUS_FILTER_ID = 'failed';
+export const SUCCESS_STATUS_FILTER_ID = 'success';
+export const BUILDING_STATUS_FILTER_ID = 'building';
+
+const RunStatusFilters = [
+  {
+    id: NEEDS_MERGE_FILTER_ID,
+    label: 'Merge build PR',
+  },
+  {
+    id: FAILED_STATUS_FILTER_ID,
+    label: 'Build failed',
+  },
+  {
+    id: SUCCESS_STATUS_FILTER_ID,
+    label: 'Build successful',
+  },
+  {
+    id: BUILDING_STATUS_FILTER_ID,
+    label: 'Building',
+  },
+];
+type ComponentFilterProps = {
+  onSelect: (selected: string[]) => void;
+  clearFilters: () => void;
+  currentFilters: string[];
+  matchCounts: {
+    [key: string]: number;
+  };
+  includeMergeStatus?: boolean;
+};
+
+const ComponentFilter: React.FC<ComponentFilterProps> = ({
+  onSelect,
+  currentFilters,
+  clearFilters,
+  matchCounts,
+  includeMergeStatus,
+}) => {
+  const [isOpen, setIsOpen] = React.useState<boolean>(false);
+
+  const onToggle = () => setIsOpen(!isOpen);
+  const onSelectItem = (event, selection) => {
+    const checked = (event.target as HTMLInputElement).checked;
+    onSelect(
+      checked ? [...currentFilters, selection] : currentFilters.filter((f) => f !== selection),
+    );
+  };
+
+  return (
+    <ToolbarGroup variant="filter-group">
+      <ToolbarFilter
+        chips={currentFilters.map((key) => ({ key, node: key }))}
+        deleteChip={(type, chip: ToolbarChip) => {
+          if (type) {
+            onSelect(currentFilters.filter((f) => f !== chip.key));
+          } else {
+            clearFilters();
+          }
+        }}
+        deleteChipGroup={() => {
+          clearFilters();
+        }}
+        categoryName="Status"
+      >
+        <Select
+          className="component-status-filter"
+          toggleAriaLabel="filter menu"
+          variant={SelectVariant.checkbox}
+          onToggle={onToggle}
+          isOpen={isOpen}
+          onSelect={onSelectItem}
+          selections={currentFilters}
+          isCheckboxSelectionBadgeHidden
```

Comment:
Environments list doesn't set `isCheckboxSelectionBadgeHidden`. We should either use it everywhere or no where.